### PR TITLE
Jade truffle  sc 9855 update jade truffle

### PR DESCRIPTION
**Pull request by JeremyGibson**
This is fairly extensive change to the template:

**Biggest Changes**
1. Sass is now a first class styling citizen
2. Node dependencies are not pinned in the template, but generated at the most current version.
3. Asset handling is moved out of `static` directories. This keeps the deployed image tidier.
4. Steps to get to a running site are reduced to four steps after the project is generated.
    - `make setup`
    - `bring up database`
    - `./manage.py migrate`
    - `make run-dev`

There are lots of other little changes:
* CSS is minified for both Sass and Tailwind
* Webpack is used for javascript
* Gulp is back as a task runner

TODO

- [x] Documentation written with the new Dev Docs in mind.
- [x] Integrate @Audiosutras and Sage's `client-onboarding` app.

### CODE REVIEW

File: hooks/post_gen_project.py [hunk 1]
```diff
@@ -1,15 +1,95 @@
...
+        "sass",
+        "gulp-sass"
+    ]
+
+    if "{{cookiecutter.css_style}}" == "tailwind":
+        base_deps += tailwind_deps
+    else:
+        base_deps += sass_deps
+
+    subprocess.run(['/bin/bash', '-i', '-c', 'nvm use'])
```
[the remark this answers is not in the document]
Comment (dchukhin) [reply hooks/post_gen_project.py:70]: As a note, adding the lines that the nvm docs recommend to put into your profile does resolve this for me:
`subprocess.run(['/bin/bash', '-i', '-c', 'export NVM_DIR="$HOME/.nvm" && [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" && nvm use'])`

**Review by dchukhin — COMMENTED** [review]
Lots of changes; good job!
I ran it locally, and almost everything worked!

It looks like some changes still need to be made to the {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/README.md, like:
 - use the user's choices of Python, NodeJS, npm versions
 - the **5. Set up local env variables** section can be removed, right?
 - we'll need to update the references to ``.env`` to be ``.envrc``

Also, 👍🏻  on the favicon 🌵

File: hooks/post_gen_project.py [hunk 2]
```diff
@@ -1,15 +1,95 @@
...
         Path(Path().cwd(), "pytest.ini").unlink()
 
+    if "{{cookiecutter.css_style}}" == "sass":
+        Path("./tailwind.config.js").unlink()
+        Path("./postcss.config.js").unlink()
+        Path("./{{cookiecutter.project_slug}}/assets/styles/tailwind_entry.css").unlink()
+        shutil.rmtree(Path("./apps/{{cookiecutter.project_slug}}/assets/styles/tailwind"))
+    else:
+        Path("./{{cookiecutter.project_slug}}/assets/styles/sass_entry.scss").unlink()
+        shutil.rmtree(Path("./apps/{{cookiecutter.project_slug}}/assets/styles/sass"))
```
Comment (kmtracey) [inline hooks/post_gen_project.py:18]: I had chosen a primary app name different from the default (and tailwind), and got an error on this `rmtree` due to the specified path not existing. Possibly this (and in the section above where tailwind stuff is removed?) should really be `{{cookiecutter.primary_app}}` (I'm not entirely sure that's the right identified.)

File: {{cookiecutter.project_slug}}/.flake8 [hunk 3]
```diff
@@ -12,3 +12,4 @@ ignore =
     E501,
     F841,   # Line too long
     E741,   # Ambiguous variable
+    W042,   # DEFAULT_AUTO_FIELD
```
Comment (kmtracey) [inline {{cookiecutter.project_slug}}/.flake8:15]: We should not encourage ignoring this warning. It's trying to get project to set an explicit value. Projects should do that, not ignore the warning. Particualrly new projects.

**Comment by kmtracey** [comment]
Dmitriy notes updating readme to "use the user's choices of Python, NodeJS, npm versions" ... but I was not asked for NodeJS nor npm versions when I used this branch so I'm confused...should I have been asked for NodeJS and npm versions? This is what I was asked and what choices I made:

```
(jade-truffle) kmtracey@caktus151 15:06:46: ~/projects/rebar
--> cookiecutter https://github.com/caktus/jade-truffle --checkout jade-truffle--sc-9855-update-jade-truffle
project_name [Apple Pie]: investigator-portal
project_slug [investigator_portal]: 
python_version [3.9.6]: 
Select project_type:
1 - wagtail
2 - django
Choose from 1, 2 [1]: 2
Select testing_type:
1 - pytest
2 - django
Choose from 1, 2 [1]: 1
Select css_style:
1 - sass
2 - tailwind
Choose from 1, 2 [1]: 2
Select ci_cd:
1 - actions
2 - circleci
Choose from 1, 2 [1]: 2
Select cloud_provider:
1 - aws
2 - gcp
Choose from 1, 2 [1]: 1
postgres_port [5432]: 
primary_app [investigator_portal]: 
project_domain_name [caktus-built.com]: 
Running post generation tasks.
```

**Comment by JeremyGibson** [comment]
Thanks for the reviews and notes. I'll look at them as address as soon as I can.

File: {{cookiecutter.project_slug}}/.flake8 (hunk 3 above)
Comment (JeremyGibson) [reply {{cookiecutter.project_slug}}/.flake8:15]: You are right, I'm going to set the DEFAULT_AUTO_FIELD in `base.py`

File: {{cookiecutter.project_slug}}/.github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          DJANGO_SETTINGS_MODULE: {{cookiecutter.project_slug}}.settings.dev
+          DATABASE_URL: postgres://postgres:postgres@127.0.0.1:5432/postgres
+  {% raw %}
+      - name: Job Notification
+        uses: 8398a7/action-slack@v3
+        with:
+          status: ${{ job.status }}
+          fields: repo,commit,ref,message,took # selectable (default: repo,message)
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }} # required
```
[the remark this answers is not in the document]
Comment (JeremyGibson) [reply {{cookiecutter.project_slug}}/.github/workflows/tests.yml:72]: Good point. That will take some thinking on how to generalize this.

**Comment by dchukhin** [comment]
@kmtracey good point. My comment was confusing; sorry about that.
It looks like I was asked the same things as you:
 - I was asked about the Python version.
 - I was not asked about the node and npm versions.
 
After going through the process of creating a project with jade-truffle, I noticed that neither the Python nor node nor npm versions in the README are correct, so we should update the README. Maybe the Python version can come from the user's Python version choice, and node version can be set the same as in the `.nvmrc` file?

File: {{cookiecutter.project_app}}/{{cookiecutter.project_app}}/settings/base.py [hunk 5]
```diff
@@ -270,13 +270,13 @@
...
-            "caktus-{{ cookiecutter.project_slug }}-team",
-            "{{ cookiecutter.project_slug }}-team@caktusgroup.com",
+            "caktus-{{ cookiecutter.project_app }}-team",
+            "{{ cookiecutter.project_app }}-team@caktusgroup.com",
         )
     ],
 )
 
 GENERAL_INQUIRY_EMAIL_RECIPIENTS = [e[1] for e in ADMINS]
 EVENT_SIGNUP_FORM_SUBMISSION_RECIPIENTS = GENERAL_INQUIRY_EMAIL_RECIPIENTS
```
Comment (kmtracey) [inline {{cookiecutter.project_app}}/{{cookiecutter.project_app}}/settings/base.py:280]: `GENERAL_INQUIRY_EMAIL_RECIPIENTS` and `EVENT_SIGNUP_FORM_SUBMISSION_RECIPIENTS` are settings that shouldn't really be in here

File: hooks/post_gen_project.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JeremyGibson) [reply hooks/post_gen_project.py]: Awesome. Thanks Dmitriy. This works perfectly.

**Pull request merged**